Core/src/IfcGeometryConverter/FaceConverter.h

```diff
@@ -22,922 +22,340 @@
-						const double nz = std::abs(normal.z);
-
-						const double nMax = std::max(std::max(nx, ny), nz);
-
-						if(nMax == nx) {
-							plane = ProjectionPlane::YZ_PLANE;
-						}
-						else if(nMax == ny) {
-							plane = ProjectionPlane::XZ_PLANE;
-						}
-						else if(nMax == nz) {
-							plane = ProjectionPlane::XY_PLANE;
-						}
-						else {
-							return false;
-						}
-					}
+				//--------------------------------------------------------------------------------------------
+				// Conversion functions
+				//--------------------------------------------------------------------------------------------
 
-					// now collect all vertices in 2D
-					//loopVertices2D.resize(loopVertices3D.size());
-
-					for(const auto& vertex : loopVertices3D) {
-						if(plane == ProjectionPlane::YZ_PLANE) {
-							loopVertices2D.push_back(carve::geom::VECTOR(vertex.y, vertex.z));
-						}
-
-						else if(plane == ProjectionPlane::XZ_PLANE) {
-							loopVertices2D.push_back(carve::geom::VECTOR(vertex.x, vertex.z));
-						}
-
-						else if(plane == ProjectionPlane::XY_PLANE) {
-							loopVertices2D.push_back(carve::geom::VECTOR(vertex.x, vertex.y));
-						}
-						else {
-							std::cout << "ERROR: plane is undefined, what??" << std::endl;
-							return false;
-						}
-					}
+				/*! \brief  Converts \c IfcBsplineSurface to ...
+					\param	surface	\c IfcBSplineSurface entity to be interpreted.
+					\param	pos
+					\param	polylineData
+					\return
+					\note	The \c IfcBSplineSurface is a subtype of \c IfcBoundedSurface.
+				*/
 
-					// check winding order of 2D polygon
-					carve::geom3d::Vector normal2D = GeomUtils::computePolygon2DNormal(loopVertices2D);
+				void convertIfcBSplineSurface(const EXPRESSReference<typename IfcEntityTypesT::IfcBsplineSurface>& surface,
+					const carve::math::Matrix& pos,
+					EXPRESSReference<carve::input::PolylineSetData>& polylineData) {
+					throw oip::UnhandledException(surface);
+				}
 
-					if(boundID == 0) {
-						if(normal2D.z < 0) {
-							std::reverse(loopVertices2D.begin(), loopVertices2D.end());
-							std::reverse(loopVertices3D.begin(), loopVertices3D.end());
-							faceLoopReversed = true;
-						}
-					}
-					else {
-						if(normal2D.z > 0) {
-							std::reverse(loopVertices2D.begin(), loopVertices2D.end());
-							std::reverse(loopVertices3D.begin(), loopVertices3D.end());
-							// faceLoopReversed = true;
-						}
-					}
+				/*! \brief  Converts \c IfcCurveBoundedPlane to ...
+					\param	surface	\c IfcCurveBoundedPlane entity to be interpreted.
+					\param	pos
+					\param	polylineData
+					\return
+					\note	The \c IfcCurveBoundedPlane is a subtype of \c IfcBoundedSurface.
+				*/
 
-					return true;
+				void convertIfcCurveBoundedPlane(const EXPRESSReference<typename IfcEntityTypesT::IfcCurveBoundedPlane>& surface,
+					const carve::math::Matrix& pos,
+					EXPRESSReference<carve::input::PolylineSetData>& polylineData) {

```

Comment:
```suggestion
					std::shared_ptr<carve::input::PolylineSetData>& polylineData
				) const throw(...)
				{
```
It's easier to read with a few more line breaks - please apply to all functions.